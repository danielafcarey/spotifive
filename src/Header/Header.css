.Header {
  text-align: right;
  padding-right: 15px; }
  .Header h1 {
    display: inline-block;
    font-size: 55px;
    margin: 0;
    padding: 10px;
    vertical-align: middle; }
  .Header img {
    display: inline-block;
    width: 65px;
    padding: 30px 6px;
    vertical-align: middle; }
  .Header .profile-photo {
    position: absolute;
    left: 10px;
    border-radius: 50%; }
