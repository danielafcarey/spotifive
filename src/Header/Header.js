import React from 'react';
import { connect } from 'react-redux';

const Header = (props) => {
  const logo = require('../images/logo.png');

  return (
    <div className='Header' >
      <img 
        className='profile-photo'
        src={ props.image !== null ? props.image : props.name } 
        alt='' />
      <h1>Spotifive</h1> 
      <img 
        className='logo'
        src={ logo } 
        alt='logo' />
    </div>
  )
};

const mapStateToProps = (state) => ({
  image: state.user.userInfo.image,
  name: state.user.userInfo.name
});

export {
  Header,
  mapStateToProps
}

export default connect(mapStateToProps)(Header);
