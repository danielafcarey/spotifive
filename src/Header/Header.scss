.Header {
  text-align: right;

  h1 {
    display: inline-block;
    font-size: 55px;
    margin: 0;
    padding: 10px;
    vertical-align: middle;
  }

  img {
    display: inline-block;
    width: 65px;
    padding: 30px 30px 30px 6px;
    vertical-align: middle;
  }
}
