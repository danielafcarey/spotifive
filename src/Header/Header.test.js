import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {

  it('matches the snapshot', () => {
    const wrapper = shallow(<Header />);

    expect(wrapper).toMatchSnapshot();
  })
})
