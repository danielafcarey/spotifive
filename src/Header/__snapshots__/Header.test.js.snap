// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Spotifive
</h1>,
        <img
          alt=""
          src="logo.png"
/>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Spotifive",
        },
        "ref": null,
        "rendered": "Spotifive",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Spotifive
</h1>,
          <img
            alt=""
            src="logo.png"
/>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Spotifive",
          },
          "ref": null,
          "rendered": "Spotifive",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
